src/learning/toolkit/rand_utils.rs

```diff
@@ -40,6 +40,41 @@ pub fn reservoir_sample<T: Copy>(pool: &[T], reservoir_size: usize) -> Vec<T> {
     res
 }
 
+///Uses a user defined StdRng generator, allowing seeded PRNG.
+/// ```
+/// use rand::{Rng, SeedableRng, StdRng};
+/// use rusty_machine::learning::toolkit::rand_utils;
+/// let seed: [_] = [1,2,3,4,5];
+/// let mut pool = &mut [1,2,3,4];
+/// let sample = rand_utils::reservoir_sample(pool, 3, seed);
+///
+/// println!("{:?}", sample);
+/// ```
+pub fn reservoir_sample_custom<T: Copy>(pool: &[T], reservoir_size: usize, seed: &[usize]) -> Vec<T> {
```

Comment:
This is a welcome change in essence - though maybe should come in a separate PR - see #138 .

However this function signature should be modified to allow a generic type implementing the [`SeedableRng`](https://doc.rust-lang.org/rand/rand/trait.SeedableRng.html). Or perhaps just take as an argument an object implementing `Rng`?